storage/src/nodestore/alloc.rs

```diff
@@ -263,32 +268,29 @@ impl FreeArea {
         address: LinearAddress,
     ) -> Result<(Self, AreaIndex), FileIoError> {
         let free_area_addr = address.get();
-        let mut stored_area_stream = storage.stream_from(free_area_addr)?;
-
-        let area_index = stored_area_stream.read_byte().map_err(|e| {
+        let stored_area_stream = storage.stream_from(free_area_addr)?;
+        Self::from_storage_reader(stored_area_stream).map_err(|e| {
             storage.file_io_error(
                 e,
                 free_area_addr,
                 Some("FreeArea::from_storage".to_string()),
             )
-        })?;
-
-        stored_area_stream
-            .next_value()
-            .map_err(|e| {
-                storage.file_io_error(
-                    e,
-                    free_area_addr,
-                    Some("FreeArea::from_storage".to_string()),
-                )
-            })
-            .map(|free_area| (free_area, area_index))
+        })
     }
 
-    pub fn as_bytes<T: ExtendableBytes>(self, prefix: u8, encoded: &mut T) {
-        encoded.push(prefix);
+    pub fn as_bytes<T: ExtendableBytes>(self, area_index: AreaIndex, encoded: &mut T) {
+        const RESERVE_SIZE: usize = size_of::<u8>() + var_int_max_size::<u64>();
+
+        encoded.reserve(RESERVE_SIZE);
+        encoded.push(area_index);
         self.write_to(encoded);
     }
+
+    fn from_storage_reader(mut reader: impl Read) -> std::io::Result<(Self, AreaIndex)> {
```

Comment:
nit: In most places in the code, this tuple is `(AreaIndex, LinearAddress)`. I wish we had a type, maybe FreeListEntry. @qusuyan runs into this a lot too.